python/lsst/ts/ess/csc/ess_csc.py

```diff
@@ -127,23 +127,19 @@ async def begin_enable(self, data: salobj.BaseDdsDataType) -> None:
         """
         await super().begin_enable(data)
         await self.cmd_enable.ack_in_progress(data, timeout=60)
+        await self.start_data_clients()
 
-    async def handle_summary_state(self) -> None:
-        if self.summary_state == salobj.State.ENABLED:
-            await self.start_data_clients()
-        else:
-            await self.stop_data_clients()
+    async def begin_disable(self, data: salobj.BaseDdsDataType) -> None:
+        await self.stop_data_clients()
 
     async def start_data_clients(self) -> None:
         """Start the data clients."""
-        # TODO DM-46349 Remove this as soon as the next XML after 22.1 is
-        #  released.
         tasks = [asyncio.create_task(client.start()) for client in self.data_clients]
         try:
             self.start_data_clients_task = asyncio.gather(*tasks)
             await self.start_data_clients_task
             self.run_data_clients_task = asyncio.create_task(self.run_data_clients())
-        except Exception as main_exception:
+        except (Exception, asyncio.CancelledError) as main_exception:
```

Comment:
Cannot that be changed to BaseException, if we need to log CancelledError? That will also log all other arithmetic etc. exceptions.